src/worker_http.rs

```diff
@@ -83,72 +82,20 @@ async fn image_payload_from_url(
     img_tfm: &Option<image_processing::ARAwareTransform>,
     aspect_ratio: &String,
     encode_images: bool,
+    img_to_rgb8: bool,
 ) -> Result<ImagePayload, image::ImageError> {
     let retries = 5;
 
     match image_from_url(client, url, retries).await {
-        Ok(mut new_image) => {
-            let original_height = new_image.height() as usize;
-            let original_width = new_image.width() as usize;
-            let mut channels = new_image.color().channel_count() as i8;
-            let mut bit_depth = (new_image.color().bits_per_pixel()
-                / new_image.color().channel_count() as u16)
-                as usize;
-
-            // Optionally transform the additional image in the same way the main image was
-            if let Some(img_tfm) = img_tfm {
-                let aspect_ratio_input = if aspect_ratio.is_empty() {
-                    None
-                } else {
-                    Some(aspect_ratio)
-                };
-
-                // TODO: tokio::spawn this
-                new_image = img_tfm
-                    .crop_and_resize(&new_image, aspect_ratio_input)
-                    .await;
-            }
-
-            let height = new_image.height() as usize;
-            let width = new_image.width() as usize;
-
-            // Encode the image if needed
-            let mut image_bytes: Vec<u8> = Vec::new();
-
-            if encode_images {
-                // If the image is 16 bits or 2 channels, standardize to 8 bits, 3 channels
-                if new_image.color() != image::ColorType::Rgb8 {
-                    new_image = image::DynamicImage::ImageRgb8(new_image.to_rgb8());
-                    bit_depth = (new_image.color().bits_per_pixel()
-                        / new_image.color().channel_count() as u16)
-                        as usize;
-                }
-
-                // Pre-encode the payload as requested, we move everything to PNGs
-                if new_image
-                    .write_to(&mut Cursor::new(&mut image_bytes), image::ImageFormat::Png)
-                    .is_err()
-                {
-                    return Err(image::ImageError::IoError(std::io::Error::new(
-                        std::io::ErrorKind::Other,
-                        "Failed to encode image",
-                    )));
-                }
-
-                channels = -1; // Signal the fact that the image is encoded
-            } else {
-                image_bytes = new_image.into_bytes();
-            }
-
-            Ok(ImagePayload {
-                data: image_bytes,
-                original_height,
-                original_width,
-                height,
-                width,
-                channels,
-                bit_depth,
-            })
+        Ok(new_image) => {
```

Comment:
refactored, both codepaths were essentially doing the same thing, now in a dedicated function